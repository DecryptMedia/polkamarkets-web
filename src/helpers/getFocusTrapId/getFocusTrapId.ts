export default function getFocusTrapId<T extends string>(
  id: T
): `focus-trap-${T}` {
  return `focus-trap-${id}`;
}
