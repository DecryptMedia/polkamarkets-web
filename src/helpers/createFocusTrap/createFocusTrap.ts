import getFocusTrapId from 'helpers/getFocusTrapId';

export default function createFocusTrap(id: string) {
  const trapEl = document.createElement('span');

  trapEl.setAttribute('tabIndex', '0');
  trapEl.dataset.testid = getFocusTrapId(id);

  return trapEl;
}
