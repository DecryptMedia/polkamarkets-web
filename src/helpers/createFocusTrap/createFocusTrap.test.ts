import getFocusTrapId from 'helpers/getFocusTrapId';

import createFocusTrap from './createFocusTrap';

enum Trap {
  ID = 'id'
}

describe('createFocusTrap', () => {
  it('returns the trapper node', () => {
    const focusTrap = createFocusTrap(Trap.ID);

    expect(focusTrap).toMatchSnapshot();
    expect(focusTrap).toHaveAttribute('data-testid', getFocusTrapId(Trap.ID));
    expect(focusTrap).toHaveAttribute('tabindex', '0');
  });
});
