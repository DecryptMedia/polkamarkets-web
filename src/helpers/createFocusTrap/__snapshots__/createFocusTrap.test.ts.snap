// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createFocusTrap returns the trapper node 1`] = `
<span
  data-testid="focus-trap-id"
  tabindex="0"
/>
`;
